File: Data Structures & Algorithms/eating-bananas/submission-1.cs
public class Solution {
    public int MinEatingSpeed(int[] piles, int h) {
        
        int l = 1;
        int r = piles.Max();
        int ans = r;
        while(l <= r) {
            int mid = l + (r - l) / 2;
            if (CanEat(mid, piles, h)) {
                ans = mid;
                r = mid - 1;
            } else {
                l = mid + 1;
            }
        }
        return ans;
    }

    private bool CanEat(int x, int[] piles, int h) {
        long sum = 0;
        foreach (int pile in piles) {
            sum += ((pile + x - 1)/x);
        }
        return sum <= h;
    }
}

File: Data Structures & Algorithms/custom-sort-string/submission-1.cpp
class Solution {
public:
    string customSortString(string order, string s) {
        string ans = "";
        vector<int> frq(27);
        for (int i = 0; i < s.size(); i++) {
            frq[s[i] - 'a']++;
        }

        for ( int i = 0; i < order.size(); i++) {
            while (frq[order[i] - 'a']) {
                ans += order[i];
                frq[order[i] - 'a']--;
            }
        }

        for (int i = 0; i < s.size(); i++) {
            while(frq[s[i] - 'a']) {
                ans += s[i];
                frq[s[i] - 'a']--;
            }
        }

        return ans;
    }
};

